pkg/obj/job.go

```diff
@@ -0,0 +1,65 @@
+package obj
+
+import (
+	"context"
+
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Job struct {
+	Key    client.ObjectKey
+	Object *batchv1.Job
+}
+
+var _ Persister = &Job{}
+var _ Loader = &Job{}
+var _ Ownable = &Job{}
+var _ LabelAnnotatableFrom = &Job{}
+
+func (j *Job) Persist(ctx context.Context, cl client.Client) error {
+	if err := Create(ctx, cl, j.Key, j.Object); err != nil {
```

Comment:
Should these be `CreateOrUpdate` instead? It feels a bit weird to have a `Create` method that will simply ignore an object that already exists -- I think I'd prefer that any logic to skip updating be part of application logic in higher-level objects or the reconciler itself.

Reply:
Jobs and PVCs are largely immutable once created. For example, with PVCs you can only update the resources after the fact. We may want to expand support later for partially updating fields that can be changed, but we'd have to customize the update functionality regardless. Conceptually, we're only interested in creating these with our specific use-case at the moment, and that's largely true in general, so it seems fairly safe to start with this baseline.